pyflakes/checker.py

```diff
@@ -1833,7 +1848,20 @@ def DICT(self, node):
     def IF(self, node):
         if isinstance(node.test, ast.Tuple) and node.test.elts != []:
             self.report(messages.IfTuple, node)
-        self.handleChildren(node)
+
+        prev = self._in_type_checking
+        if _is_typing(node.test, 'TYPE_CHECKING', self.scopeStack):
+            self._in_type_checking = True
+        # Handle the body of this if statement.
+        body_nodes = node.body
```

Comment:
We could also add another argument only=["body"] to handleChildren to make it clearer.